// cDB/cdb.js
var cSettings = require('cSettings');
var db = require('mongoskin').db(cSettings.db.mongolab.url);

exports.route = [{
   api: 'db',
   method: 'GET',
   callback: get
}, {
   api: 'db',
   method: 'POST',
   callback: post
}, {
   api: 'db',
   method: 'PUT',
   callback: put
}, {
   api: 'db',
   method: 'DELETE',
   callback: del
}];

function get(req, res) {
   parse(req, res, _get);
}

function post(req, res) {
   parse(req, res, _post);
}

function put(req, res) {
   parse(req, res, _put);
}

function del(req, res) {
   parse(req, res, _del);
}

function parse(req, res, callback) {
   var collection = req.cURL.path[0];
   if (collection === 'users') {
      res.end("Not allowed"); // TODO: route to 404 page
      return;
   }
   callback(req, res, collection);
}

function _get(req, res, coll) {
   db.collection(coll).find({
      "user": req.user.id
   }).toArray(function (err, elements) {
      if (err) res.end(JSON.stringify(err));
      else res.write(JSON.stringify(elements));
      res.end();
   });
}

function _post(req, res, coll) {
   req.post.user = req.user.id;
   db.collection(coll).insert(req.post, function (err, replies) {
      if (err) res.end(JSON.stringify(err));
      else res.end('OK');
   });
}

function _put(req, res, coll) {}

function _del(req, res, coll) {}

exports.get = get;
exports.post = post;
exports.put = put;
exports.del = del;
